fdbserver/storageserver.actor.cpp

```diff
@@ -13501,6 +13516,32 @@ ACTOR Future<Void> storageServerCore(StorageServer* self, StorageServerInterface
 				updateProcessStatsTimer = delay(SERVER_KNOBS->FASTRESTORE_UPDATE_PROCESS_STATS_INTERVAL);
 			}
 			when(wait(self->actors.getResult())) {}
+			when(GetBusyShardsRequest req = waitNext(ssi.getBusyShards.getFuture())) {
+				std::vector<std::pair<KeyRange, int64_t>> topRanges;
+				for (auto& s : self->shards.ranges()) {
+					KeyRange keyRange = KeyRange(s.range());
+					int64_t total = self->metrics.getHotShards(keyRange);
+
+					if (topRanges.size() < SERVER_KNOBS->SHARD_THROTTLING_TRACKED) {
+						topRanges.emplace_back(keyRange, total);
+					} else if (total > topRanges.back().second) {
+						topRanges.pop_back();
+						topRanges.emplace_back(keyRange, total);
+					}
```

Comment:
Not sure this algorithm can select top k hottest shards. Suppose the input is 1,1,2,3,4, and the k=2, the algorithm seems to return 1,4. However, we actual want 3,4. May be using a priority queue here.

Reply:
yes, It should be a priority queue